pkg/tagger/tagstore_test.go

```diff
@@ -216,20 +229,46 @@ func TestDuplicateSourceTags(t *testing.T) {
 	etags.highCardTags["sourceNodeOrchestrator"] = []string{"tag3:sourceHigh", "tag4:sourceHigh"}
 	etags.highCardTags["sourceClusterOrchestrator"] = []string{"tag4:sourceClusterLow"}
 	etags.lowCardTags["sourceClusterOrchestrator"] = []string{"tag3:sourceClusterHigh", "tag1:sourceClusterLow"}
-	tags, sources = etags.get(true)
+	tags, sources, hash = etags.get(true)
 	assert.Len(t, tags, 0)
 	assert.Len(t, sources, 0)
 	assert.True(t, etags.cacheValid)
+	assert.Empty(t, hash)
 
 	// Invalidate the cache, we should now get the tags
 	etags.cacheValid = false
-	tags, sources = etags.get(true)
+	tags, sources, hash = etags.get(true)
 	assert.Len(t, tags, 7)
 	assert.ElementsMatch(t, tags, []string{"foo", "bar", "tag1:sourceClusterLow", "tag2:sourceHigh", "tag3:sourceClusterHigh", "tag4:sourceClusterLow", "tag5:sourceLow"})
 	assert.Len(t, sources, 3)
 	assert.True(t, etags.cacheValid)
-	tags, sources = etags.get(false)
+	assert.Equal(t, "b4e89f91534288c8", hash)
+	tags, sources, hash = etags.get(false)
 	assert.Len(t, sources, 3)
 	assert.Len(t, tags, 5)
 	assert.ElementsMatch(t, tags, []string{"foo", "bar", "tag1:sourceClusterLow", "tag2:sourceHigh", "tag3:sourceClusterHigh"})
+	assert.Equal(t, "b4e89f91534288c8", hash)
+}
+
+func shuffleTags(tags []string) {
+	for i := range tags {
+		j := rand.Intn(i + 1)
+		tags[i], tags[j] = tags[j], tags[i]
+	}
+}
+
+func TestDigest(t *testing.T) {
+	tags := []string{
```

Comment:
Let's add two `high1` to check the digest issue

Reply:
I added some tags in the test